pkgs/by-name/st/stash/package.nix

```diff
@@ -1,50 +1,147 @@
 {
+  buildGoModule,
+  darwin,
+  fetchFromGitHub,
+  fetchYarnDeps,
   lib,
+  nixosTests,
+  nodejs,
   stdenv,
-  fetchurl,
+  testers,
+  yarnBuildHook,
+  yarnConfigHook,
 }:
 let
+  inherit (lib.importJSON ./version.json)
+    gitHash
+    srcHash
+    vendorHash
+    version
+    yarnHash
+    ;
 
-  version = "0.25.1";
+  pname = "stash";
 
-  sources = {
-    x86_64-linux = {
-      url = "https://github.com/stashapp/stash/releases/download/v${version}/stash-linux";
-      hash = "sha256-Rb4x6iKx6T9NPuWWDbNaz+35XPzLqZzSm0psv+k2Gw4=";
-    };
-    aarch64-linux = {
-      url = "https://github.com/stashapp/stash/releases/download/v${version}/stash-linux-arm64v8";
-      hash = "sha256-6qPyIYKFkhmBNO47w9E91FSKlByepBOnl0MNJighGSc=";
-    };
-    x86_64-darwin = {
-      url = "https://github.com/stashapp/stash/releases/download/v${version}/stash-macos";
-      hash = "sha256-W8+rgqWUDTOB8ykGO2GL9tKEjaDXdx9LpFg0TAtJsxM=";
-    };
+  src = fetchFromGitHub {
+    owner = "stashapp";
+    repo = "stash";
+    rev = "refs/tags/v${version}";
+    hash = srcHash;
   };
-in
-stdenv.mkDerivation (finalAttrs: {
-  inherit version;
 
-  pname = "stash";
+  ui = stdenv.mkDerivation (final: {
+    inherit version;
+    pname = "${pname}-ui";
+    src = "${src}/ui/v2.5";
+
+    yarnOfflineCache = fetchYarnDeps {
+      yarnLock = "${final.src}/yarn.lock";
+      hash = yarnHash;
+    };
+
+    nativeBuildInputs = [
+      yarnConfigHook
```

Comment:
Build fails on Darwin.
```
Executing yarnConfigHook
yarn config v1.22.22
error Error: EEXIST: file already exists, mkdir '/nix/store/ha7f20d8g038gpzlmklcjgfgbblbg6fs-offline'
info Visit https://yarnpkg.com/en/docs/cli/config for documentation about this command.
```

Reply:
I don't know what's happening there. I don't own any physical mac hardware but using OSX-KVM with MacOS Ventura, I was able to build and run.

Reply:
I think this was also fixed for all yarnConfigHook uses meanwhile.